### PR TITLE
Fix can't get pfname for virtio pci path

### LINKED ISSUE #507: use usespace vdpa can't find the pfName because the net path is wrong
### What happened?
 There is no net path  in Virtio-net is the net layer of virtio, and the standard kernel driver places this layer separately into the virtio1 path
e74c25da4e1dc57466fd8e40174e5812

### What did you expect to happen?
If the net folder is not found in the **/sys/bus/pci/devices/pciAddr/physfn/net** path, I hope you can find the folder starting with virtio again
### What are the minimal steps needed to reproduce the bug?
No**/sys/bus/pci/devices/pciAddr/physfn/net**
Only the **/sys/bus/pci/devices/pciAddr virtioxx/physfn/net**
### Anything else we need to know?
0fcfcfcf1d5fd87ff5f527946522b39c
a80b6274adbf5a1d9f766357260cebd0
5bed714b280cbf48fc35600bc7e7532d

### Component Versions
Please fill in the below table with the version numbers of components used.

Component | Version|
------------------------------|--------------------|
|SR-IOV Network Device Plugin |0bf8f4fa9dc4ef6a0b88c875142815676d3bf311|
|SR-IOV CNI Plugin            |0bf8f4fa9dc4ef6a0b88c875142815676d3bf311|
|Multus                       ||
|Kubernetes                   |v1.24.0| 
|OS                           |ubuntu 22.04|

### Config Files
Config file locations may be config dependent.

##### SR-IOV Network Device Plugin Logs (use `kubectl logs $PODNAME`)
root@master:~/temp/kubevirt# kubectl logs -f -n kube-system      kube-sriov-device-plugin-amd64-d6vsq
I1108 14:43:33.442520       1 manager.go:51] Using Kubelet Plugin Registry Mode
I1108 14:43:33.442648       1 main.go:44] resource manager reading configs
I1108 14:43:33.442685       1 manager.go:79] raw ResourceList: {
  "resourceList": [{
        "resourceName": "sriov_dpu",
        "resourcePrefix": "yusur.tech",
        "selectors": {
                "rootDevices": ["0000:01:00.0#1-6"]
        }
      }
  ]
}
I1108 14:43:33.442759       1 factory.go:166] net device selector for resource sriov_dpu is &{DeviceSelectors:{Vendors:[] Devices:[] Drivers:[] PciAddresses:[]} PfNames:[] RootDevices:[0000:01:00.0#1-6] LinkTypes:[] DDPProfiles:[] IsRdma:false NeedVhostNet:false}
I1108 14:43:33.442781       1 manager.go:99] unmarshalled ResourceList: [{ResourcePrefix:yusur.tech ResourceName:sriov_dpu DeviceType:netDevice Selectors:0xc00000cd38 SelectorObj:0xc0001b32b0}]
I1108 14:43:33.442815       1 manager.go:200] validating resource name "yusur.tech/sriov_dpu"
I1108 14:43:33.442818       1 main.go:60] Discovering host devices
I1108 14:43:33.469296       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0a.0 02              unknown                 unknown
I1108 14:43:33.469535       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0a.1 02              unknown                 unknown
I1108 14:43:33.469605       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0a.2 02              unknown                 unknown
I1108 14:43:33.469659       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0a.3 02              unknown                 unknown
I1108 14:43:33.469719       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0a.4 02              unknown                 unknown
I1108 14:43:33.469767       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0a.5 02              unknown                 unknown
I1108 14:43:33.469814       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0a.6 02              unknown                 unknown
I1108 14:43:33.469875       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0a.7 02              unknown                 unknown
I1108 14:43:33.469931       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0b.0 02              unknown                 unknown
I1108 14:43:33.469990       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:00:0c.0 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470009       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:01:00.0 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470035       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:01:00.1 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470045       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:01:00.2 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470054       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:01:00.3 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470064       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:01:00.4 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470075       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:01:00.5 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470086       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:01:00.6 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470097       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:01:00.7 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470106       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:02:00.0 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470127       1 netDeviceProvider.go:84] netdevice AddTargetDevices(): device found: 0000:10:00.0 02              Red Hat, Inc.           Virtio network device
I1108 14:43:33.470151       1 main.go:66] Initializing resource servers
I1108 14:43:33.470157       1 manager.go:105] number of config: 1
I1108 14:43:33.470160       1 manager.go:109]
I1108 14:43:33.470162       1 manager.go:110] Creating new ResourcePool: sriov_dpu
I1108 14:43:33.470166       1 manager.go:111] DeviceType: netDevice
I1108 14:43:33.470339       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0a.1. error getting devlink device attributes for net device 0000:00:0a.0 no such device
I1108 14:43:33.470511       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0a.2. error getting devlink device attributes for net device 0000:00:0a.0 no such device
I1108 14:43:33.470694       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0a.3. error getting devlink device attributes for net device 0000:00:0a.0 no such device
I1108 14:43:33.471309       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0a.4. error getting devlink device attributes for net device 0000:00:0a.0 no such device
I1108 14:43:33.471644       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0a.5. error getting devlink device attributes for net device 0000:00:0a.0 no such device
I1108 14:43:33.471959       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0a.6. error getting devlink device attributes for net device 0000:00:0a.0 no such device
I1108 14:43:33.472276       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0a.7. error getting devlink device attributes for net device 0000:00:0a.0 no such device
I1108 14:43:33.472473       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0b.0. error getting devlink device attributes for net device 0000:00:0b.0 no such device
I1108 14:43:33.472583       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:00:0c.0. error getting devlink device attributes for net device 0000:00:0c.0 no such device
W1108 14:43:33.472592       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:00:0c.0/net: no such file or directory"
I1108 14:43:33.472827       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:01:00.1. error getting devlink device attributes for net device 0000:01:00.0 no such device
W1108 14:43:33.472837       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:01:00.1/net: no such file or directory"
I1108 14:43:33.472977       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:01:00.2. error getting devlink device attributes for net device 0000:01:00.0 no such device
W1108 14:43:33.472984       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:01:00.2/net: no such file or directory"
I1108 14:43:33.473145       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:01:00.3. error getting devlink device attributes for net device 0000:01:00.0 no such device
W1108 14:43:33.473154       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:01:00.3/net: no such file or directory"
I1108 14:43:33.473338       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:01:00.4. error getting devlink device attributes for net device 0000:01:00.0 no such device
W1108 14:43:33.473347       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:01:00.4/net: no such file or directory"
I1108 14:43:33.473572       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:01:00.5. error getting devlink device attributes for net device 0000:01:00.0 no such device
W1108 14:43:33.473580       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:01:00.5/net: no such file or directory"
I1108 14:43:33.473825       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:01:00.6. error getting devlink device attributes for net device 0000:01:00.0 no such device
W1108 14:43:33.473834       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:01:00.6/net: no such file or directory"
I1108 14:43:33.474096       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:01:00.7. error getting devlink device attributes for net device 0000:01:00.0 no such device
W1108 14:43:33.474104       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:01:00.7/net: no such file or directory"
I1108 14:43:33.474171       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:02:00.0. error getting devlink device attributes for net device 0000:02:00.0 no such device
W1108 14:43:33.474178       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:02:00.0/net: no such file or directory"
I1108 14:43:33.474239       1 utils.go:71] Devlink query for eswitch mode is not supported for device 0000:10:00.0. error getting devlink device attributes for net device 0000:10:00.0 no such device
W1108 14:43:33.474246       1 pciNetDevice.go:81] unable to get PF name "open /sys/bus/pci/devices/0000:10:00.0/net: no such file or directory"
I1108 14:43:33.474261       1 factory.go:106] device added: [pciAddr: 0000:01:00.2, vendor: 1af4, device: 1041, driver: virtio-pci]
I1108 14:43:33.474263       1 factory.go:106] device added: [pciAddr: 0000:01:00.3, vendor: 1af4, device: 1041, driver: virtio-pci]
I1108 14:43:33.474265       1 factory.go:106] device added: [pciAddr: 0000:01:00.4, vendor: 1af4, device: 1041, driver: virtio-pci]
I1108 14:43:33.474267       1 factory.go:106] device added: [pciAddr: 0000:01:00.5, vendor: 1af4, device: 1041, driver: virtio-pci]
I1108 14:43:33.474271       1 factory.go:106] device added: [pciAddr: 0000:01:00.6, vendor: 1af4, device: 1041, driver: virtio-pci]
I1108 14:43:33.474273       1 factory.go:106] device added: [pciAddr: 0000:01:00.7, vendor: 1af4, device: 1041, driver: virtio-pci]
I1108 14:43:33.474281       1 manager.go:139] New resource server is created for sriov_dpu ResourcePool
I1108 14:43:33.474285       1 main.go:72] Starting all servers...
I1108 14:43:33.474446       1 server.go:199] starting sriov_dpu device plugin endpoint at: yusur.tech_sriov_dpu.sock
I1108 14:43:33.474818       1 server.go:226] sriov_dpu device plugin endpoint started serving
I1108 14:43:33.474842       1 main.go:77] All servers started.
I1108 14:43:33.474848       1 main.go:78] Listening for term signals
I1108 14:43:33.964954       1 server.go:110] Plugin: yusur.tech_sriov_dpu.sock gets registered successfully at Kubelet
I1108 14:43:33.964962       1 server.go:134] ListAndWatch(sriov_dpu) invoked
I1108 14:43:33.964969       1 server.go:142] ListAndWatch(sriov_dpu): send devices &ListAndWatchResponse{Devices:[]*Device{&Device{ID:0000:01:00.6,Health:Healthy,Topology:nil,},&Device{ID:0000:01:00.7,Health:Healthy,Topology:nil,},&Device{ID:0000:01:00.2,Health:Healthy,Topology:nil,},&Device{ID:0000:01:00.3,Health:Healthy,Topology:nil,},&Device{ID:0000:01:00.4,Health:Healthy,Topology:nil,},&Device{ID:0000:01:00.5,Health:Healthy,Topology:nil,},},}

### PR DESCRIPTION
**Pull request by JasinlikeEatingOrange**
Fixed issue with virtio driver by adding a conditional check for the virtio path under the PCI address, preventing errors when the pf name cannot be found. This ensures successful acquisition of vf resources.

Fixes #507

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -92,6 +92,17 @@ func GetPfName(pciAddr string) (string, error) {
 	}
 
 	path := filepath.Join(sysBusPci, pciAddr, "physfn", "net")
+	vertIOpath := filepath.Join(sysBusPci, pciAddr, "physfn")
+	dir, err := os.ReadDir(vertIOpath)
```
Comment (zeeke) [inline pkg/utils/utils.go:96]: nit:
```suggestion
	dir, err := os.ReadDir(virtIOpath)
```

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -92,6 +92,17 @@ func GetPfName(pciAddr string) (string, error) {
...
+	if err != nil {
+		return "", nil
```
Comment (zeeke) [inline pkg/utils/utils.go:98]: Maybe we want to return the error
```suggestion
		return "", fmt.Errorf("can't open path %s: %w", virtIOpath, err)
```

**Review by zeeke — COMMENTED** [review]
Left a few comment. Not an expert of virtIO, but is there any chance to cover this change via unit tests? See [utils_test.go](https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/blob/master/pkg/utils/utils_test.go#L405)

**Comment by JasinlikeEatingOrange** [comment]
> Left a few comment. Not an expert of virtIO, but is there any chance to cover this change via unit tests? See [utils_test.go](https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/blob/master/pkg/utils/utils_test.go#L405)

sorry, i have modify the code and add test case, it passed . thx for review.

**Comment by SchSeba** [comment]
can I close this issue?

**Comment by SchSeba** [comment]
closing

**Pull request closed without merging**